lib/web_ui/lib/src/engine/window.dart

```diff
@@ -198,6 +203,50 @@ class EngineWindow extends ui.Window {
       callback(data);
     });
   }
+
+
+  /// The setting indicating the current brightness mode of the host platform.
+  /// If the platform has no preference, [platformBrightness] defaults to [Brightness.light].
+  ui.Brightness get platformBrightness => _platformBrightness;
+  ui.Brightness _platformBrightness = ui.Brightness.light;
+
+  /// Reference to css media query that indicates the user theme preference on the web
+  final html.MediaQueryList _brightnessMediaQuery = html.window.matchMedia('(prefers-color-scheme: dark)');
+
+  /// A callback that is invoked whenever [_brightnessMediaQuery] changes value.
+  /// Updates the platformBrightness with the new user preferenceq
+  ///
+  ///  The param brightnessMediaQuery can be a MediaQueryListEvent or a MediaQueryList;
+  ///
+  Function(dynamic) _brightnessMediaQueryListener;
+
+  /// Set the callback function for listening changes in [_brightnessMediaQuery] value
+  void _addBrightnessMediaQueryListener() {
+    _brightnessMediaQueryListener = (brightnessMediaQuery) {
+       ui.Brightness previousPlatformBrightness = _platformBrightness;
+      _platformBrightness = brightnessMediaQuery.matches ? ui.Brightness.dark : ui.Brightness.light;
```

Comment:
We obsessively avoid dynamic dispatch. Let's downcast to `MediaQueryListEvent` before accessing members.